registry/client/auth/session.go

```diff
@@ -209,7 +210,7 @@ func (th *tokenHandler) fetchToken(params map[string]string) (token string, err
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
+	if !client.SuccessStatus(resp.StatusCode) {
```

Comment:
I think this should just do the check like transport does, then `SuccessStatus` does not have to be exported.